### DISCUSSION TITLE
Drawer

### DISCUSSION THREAD

**Discussion by jeffreylauwers** [nl-design-system/backlog#238] [Component suggestions]
<img width="720" alt="visuele weergave van het drawer component" src="https://github.com/nl-design-system/backlog/assets/545207/a4810e46-31ba-48db-a4bb-a562529f4d5d">

We willen graag meer leren over de **Drawer** component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook of screenshots. Met deze informatie kunnen we bewijzen dat het component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1712824139211439)

---

## Naam
Drawer

## Doel
Een paneel aan de zijkant van een scherm, die geopend kan worden voor gerelateerde informatie, formulieren of acties.

## Gerelateerde componenten
[Modal Dialog](https://github.com/orgs/nl-design-system/discussions/237), [Non Modal Dialog](https://github.com/orgs/nl-design-system/discussions/280)

## Onderzoek
Niet veel onderzoek kunnen vinden rondom het Drawer component.

[Bottom Sheets: Definition and UX Guidelines -  Nielsen Norman Group](https://www.nngroup.com/articles/bottom-sheet/)
Artikel dat ingaat op 'Bottom Sheets'. Een component dat gelijkenis vertoond. Zeker vanwege de vele alignment opties bij enkele uitvoeringen.

[UX designers: Side drawer navigation could be costing you half your user engagement - The Next Web](https://thenextweb.com/news/ux-designers-side-drawer-navigation-costing-half-user-engagement)
Artikel over een gebruikersonderzoek naar de zijnavigatie in een applicatie. Deze zijnavigatie bleek een negatief effect te hebben op de gebruikersbetrokkenheid. Engagement, inclusief de tijd die in de app werd doorgebracht, werd gehalveerd.

[Bekijk het Modal Dialog component voor meer informatie](https://github.com/orgs/nl-design-system/discussions/237)

**Comment by jeffreylauwers** [comment]
## Amsterdam

In [Figma](https://www.figma.com/file/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?type=design&node-id=12031-2567&mode=design&t=qJUOy71XQN5GGKKD-4) onder Filter pattern. Waar verwezen wordt naar een [Figma prototype](https://www.figma.com/proto/QnhEIYb52XF1EcbOZdKvgJ/ERISA-(WIP)?page-id=2715%3A73578&type=design&node-id=2719-74049&viewport=18632%2C1898%2C0.25&t=E5QLnQqZZCtRkkP1-1&scaling=scale-down&starting-point-node-id=2719%3A74049&mode=design)

---

### Screenshots / Varianten

**Figma**

Closed

![Scherm­afbeelding 2024-04-11 om 14 24 46](https://github.com/nl-design-system/backlog/assets/545207/6d48fed8-2d7b-4107-bf86-4788a345f85b)

Open

![Scherm­afbeelding 2024-04-11 om 14 25 41](https://github.com/nl-design-system/backlog/assets/545207/594af418-c84e-4e74-8170-57f70219cead)

![Scherm­afbeelding 2024-04-11 om 14 25 52](https://github.com/nl-design-system/backlog/assets/545207/14b4350e-9c29-4df8-9a0b-b4595bb86b69)

---

### Notities
- Niet als los component, maar als onderdeel van filter patroon.
- Sluit optie en button group positie gelijk aan Modal dialog component.
- Backdrop aanwezig.

[Bekijk het Modal dialog component van Amsterdam voor meer informatie](https://github.com/orgs/nl-design-system/discussions/237#discussioncomment-9021935)

**Comment by jeffreylauwers** [comment]
## Den Haag

- [Figma - Huidig](https://www.figma.com/file/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?type=design&node-id=1713-5970&mode=design&t=DHGSgFL21Y27ZSKX-4)
- [Figma - WIP](https://www.figma.com/file/x4RkF6BIdrNZbh7D53NTzB/%F0%9F%92%A0-NLDS---Den-Haag---Bibliotheek?type=design&node-id=3420-183305&mode=design&t=YT09XiO5K5CN9PQF-4)

---

### Screenshots / Varianten

**Figma Huidig**

Default

![Scherm­afbeelding 2024-04-11 om 14 38 21](https://github.com/nl-design-system/backlog/assets/545207/660c43cf-e1a1-4e72-adcc-d09484c393e6)

**Figma WIP**

Default

![Scherm­afbeelding 2024-04-11 om 14 39 29](https://github.com/nl-design-system/backlog/assets/545207/bd7fd867-9053-4f1a-806c-fbd8e2a89b75)

With footer

![Scherm­afbeelding 2024-04-11 om 14 39 37](https://github.com/nl-design-system/backlog/assets/545207/6c63e3c1-b41d-4a3d-98df-56e4923e824b)

Align: Inline end

![Scherm­afbeelding 2024-04-11 om 14 39 45](https://github.com/nl-design-system/backlog/assets/545207/8da6a5e6-3802-4bfb-a13a-1169345d0ac9)

Align: Block end

![Scherm­afbeelding 2024-04-11 om 14 39 51](https://github.com/nl-design-system/backlog/assets/545207/220392b0-5c26-4a9d-96ff-8cb4e014df4f)

---

### Notities
- Enkel in Figma beschikbaar.
- Richtlijnen beschikbaar (Figma Huidig).
- Opties voor Footer en Alignment beschikbaar (Figma WIP)
- Heading kleur verschilt tussen Drawer en Modal dialog (Figma Huidig)
- Sluit optie en button group positie gelijk aan Modal dialog component (Figma WIP).
- Geen Backdrop aanwezig.

[Bekijk het Modal dialog component van Den Haag voor meer informatie](https://github.com/orgs/nl-design-system/discussions/237#discussioncomment-9022565)

**Comment by jeffreylauwers** [comment]
## DUO

[Documentatie website](https://uno.dfront.rijkscloud.nl/#/components/drawer)

---

### Screenshots / Varianten

**Documentatie website**

Default

![uno dfront rijkscloud nl_](https://github.com/nl-design-system/backlog/assets/545207/c10382ea-0424-4107-b801-86add43e9bc5)

Small viewport

![Scherm­afbeelding 2024-04-11 om 14 59 23](https://github.com/nl-design-system/backlog/assets/545207/4623c6dc-1a07-48dc-b9f5-5a9db866ff0d)


Drawer met backdrop
Met behulp van de .modal--backdrop class kan de drawer voorzien worden van den backdrop, die ook alle elementen achter de drawer uitschakeld. Deze moet gezet worden op dezelfde div als de .modal class.

![uno dfront rijkscloud nl_ (1)](https://github.com/nl-design-system/backlog/assets/545207/e4988146-3e91-48f6-9d32-f3bc4a2626f9)

Small viewport

![Scherm­afbeelding 2024-04-11 om 14 59 23](https://github.com/nl-design-system/backlog/assets/545207/4623c6dc-1a07-48dc-b9f5-5a9db866ff0d)

---

### Notities
- Richtlijnen beschikbaar.
- Backdrop als optie beschikbaar.
- Bij small viewport vult Drawer gehele viewport.
- Opmaak gelijk aan Modal dialog component.

[Bekijk het Modal dialog component van DUO voor meer informatie](https://github.com/orgs/nl-design-system/discussions/237#discussioncomment-9022610)

**Comment by jeffreylauwers** [comment]
## Rotterdam

In [Figma](https://www.figma.com/file/ZWSC4gCrOXRUR9UX3aoZ8x/Gemeente-Rotterdam-Design-System?type=design&node-id=516-21602&mode=design&t=Aao1HMrV50O4Jd9S-4) onder Tablet navigation. 

---

### Screenshots / Varianten

**Figma**

Tablet navigation

![Scherm­afbeelding 2024-04-11 om 15 04 00](https://github.com/nl-design-system/backlog/assets/545207/4702422a-e1a6-42a6-82a5-a0ddc3c360d5)

---

### Notities
- Niet als los component, maar als onderdeel van navigation bij medium viewport.
- Sluit optie gelijk aan Modal dialog component.
- Backdrop aanwezig.

[Bekijk het Modal dialog component van Rotterdam voor meer informatie](https://github.com/orgs/nl-design-system/discussions/237#discussioncomment-9042393)

**Comment by jeffreylauwers** [comment]
## RVO

- [Documentatie website](https://nl-design-system.github.io/rvo/docs/components/menubar-mobile/) onder Menubar (mobile)
- [Storybook](https://nl-design-system.github.io/rvo/?path=/docs/componenten-mobile-menubar--documentatie) onder Mobile Menubar.
- [GitHub](https://github.com/nl-design-system/rvo/tree/master/components/menubar-mobile) onder Mobile Menubar.
---

### Screenshots / Varianten

**Storybook**

Closed

![Scherm­afbeelding 2024-04-11 om 15 13 27](https://github.com/nl-design-system/backlog/assets/545207/beeca3f9-91e3-4949-a59b-73ae7091ed71)

Open

![Scherm­afbeelding 2024-04-11 om 15 13 16](https://github.com/nl-design-system/backlog/assets/545207/06773536-f011-414d-89bd-64cc037602e3)

---

### Notities
- Component lijkt enkel in code beschikbaar onder Menubar (mobile).
- Wijkt behoorlijk af van Modal dialog.
- Backdrop aanwezig.

[Bekijk het Modal dialog component van RVO voor meer informatie](https://github.com/orgs/nl-design-system/discussions/237#discussioncomment-9042447)

**Comment by jeffreylauwers** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-drawer--docs)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/drawer)
- [Figma](https://www.figma.com/file/UXIHcIurAD8hyoBWx4hDBV/NLDS---Gemeente-Utrecht---Bibliotheek?type=design&node-id=3113-27967&mode=design&t=Ywws49XSeWT6uDPh-4)

---

### Screenshots / Varianten

**Storybook**

Default

![Scherm­afbeelding 2024-04-11 om 15 19 26](https://github.com/nl-design-system/backlog/assets/545207/eea4cbce-32d9-4f60-b556-77c2d32d0b6c)

inline-start alignment

![Scherm­afbeelding 2024-04-11 om 15 21 06](https://github.com/nl-design-system/backlog/assets/545207/8ef4da01-ebdb-4539-a963-b15396e9f029)

inline-end alignment

![Scherm­afbeelding 2024-04-11 om 15 21 13](https://github.com/nl-design-system/backlog/assets/545207/a2a978a2-53f7-4ba4-94eb-397eb60d41ec)

block-start alignment

![Scherm­afbeelding 2024-04-11 om 15 21 19](https://github.com/nl-design-system/backlog/assets/545207/dd29d99e-bfbe-4c45-a427-1a592b754e7b)

block-end alignment

![Scherm­afbeelding 2024-04-11 om 15 21 24](https://github.com/nl-design-system/backlog/assets/545207/acebf7c3-db4b-43ec-a1c9-a5dc610b01c0)

Overflow Y

![Scherm­afbeelding 2024-04-11 om 15 21 30](https://github.com/nl-design-system/backlog/assets/545207/623e9799-675b-4b76-a376-fcdb6ce06391)

**Figma**

Default

![Scherm­afbeelding 2024-04-11 om 15 21 51](https://github.com/nl-design-system/backlog/assets/545207/fef439a7-fb76-482a-90af-73903e1621ff)

With footer

![Scherm­afbeelding 2024-04-11 om 15 22 01](https://github.com/nl-design-system/backlog/assets/545207/a0572f9a-8431-4924-8ea1-9584be0562e5)

Align: Inline end

![Scherm­afbeelding 2024-04-11 om 15 22 11](https://github.com/nl-design-system/backlog/assets/545207/b91bfdb7-eee1-4297-8224-ca3975ab329f)

Align: Block end

![Scherm­afbeelding 2024-04-11 om 15 22 17](https://github.com/nl-design-system/backlog/assets/545207/99c456ad-a720-4128-99ac-a166a809f14d)

---

### Notities

- Veel verschillende opties voor alignment beschikbaar.
- Storybook component wijkt behoorlijk af van Modal dialog.
- Opmaak gelijk aan Modal dialog component.
- Geen backdrop aanwezig.

[Bekijk het Modal dialog component van Utrecht voor meer informatie](https://github.com/orgs/nl-design-system/discussions/237#discussioncomment-9046733)

**Comment by JJGrootveld** [comment]
# Voorbeelden KVK.nl

 **Let op:**Dit component zit nog niet in het KVK Design System. Dit betekend dat er nog verschillende vormen van op de website leven, maar ik heb dmv notes geprobeerd het gedrag te omschrijven.

## Artikeloverzicht
https://www.kvk.nl/overzicht/

### Desktop
<img width="1787" alt="Artikelen01" src="https://github.com/nl-design-system/backlog/assets/9549605/a99eb441-e98a-4482-9740-9883e486c83c">
<img width="1792" alt="Artikelen02" src="https://github.com/nl-design-system/backlog/assets/9549605/32c96bb7-3b5e-4f94-8c14-d9cdf36634b4">
<img width="1822" alt="Artikelen03" src="https://github.com/nl-design-system/backlog/assets/9549605/8cd2b59e-701d-472d-81f0-31064cff4beb">
<img width="1751" alt="Artikelen04" src="https://github.com/nl-design-system/backlog/assets/9549605/1cfa94ae-34cc-4bd5-831e-d5cc4e3836ad">

### Mobile
<img width="728" alt="Artikelen01 - Mob" src="https://github.com/nl-design-system/backlog/assets/9549605/831ac7d1-6283-44ac-91ac-f2daa3ac069b">
<img width="724" alt="Artikelen02 - Mob" src="https://github.com/nl-design-system/backlog/assets/9549605/aa4454f0-74d3-4eb8-83c5-3b9d831ef393">
<img width="662" alt="Artikelen03 - Mob" src="https://github.com/nl-design-system/backlog/assets/9549605/5ab22fd1-e5a0-416e-933d-0603e0008fcb">
<img width="716" alt="Artikelen04 - Mob" src="https://github.com/nl-design-system/backlog/assets/9549605/d4dc617e-8973-4a14-ae6a-45038d00b68d">

### Zoeken

https://www.kvk.nl/zoeken/

<img width="1784" alt="Zoeken01" src="https://github.com/nl-design-system/backlog/assets/9549605/9ddf5de7-ca35-4843-b67c-431e56eba9bd">
<img width="1786" alt="Zoeken02" src="https://github.com/nl-design-system/backlog/assets/9549605/ac14c490-7f19-4174-9251-ffbbd66c52b1">

**Comment by Rozerinay** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Drawer op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=drawer)
